scripts/install_third_party_libs.py

```diff
@@ -0,0 +1,253 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Installation script for Oppia third-party libraries."""
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+
+import argparse
+import contextlib
+import fileinput
+import os
+import shutil
+import subprocess
+import sys
+
+import python_utils
+
+from . import common
+from . import install_third_party
+from . import pre_commit_hook
+from . import pre_push_hook
+from . import setup
+
+_PARSER = argparse.ArgumentParser()
+_PARSER.add_argument(
+    '--nojsrepl',
+    help='optional; if specified, skips installation of skulpt.',
+    action='store_true')
+_PARSER.add_argument(
+    '--noskulpt',
+    help='optional; if specified, skips installation of skulpt.',
+    action='store_true')
+
+
+@contextlib.contextmanager
+def _redirect_stdout(new_target):
+    """Redirect stdout to the new target.
+
+    Args:
+        new_target: TextIOWrapper. The new target to which stdout is redirected.
+
+    Yields:
+        TextIOWrapper. The new target.
+    """
+    old_target = sys.stdout
+    sys.stdout = new_target
+    try:
+        yield new_target
+    finally:
+        sys.stdout = old_target
+
+
+def pip_install(package, version, install_path):
+    """Installs third party libraries with pip.
+
+    Args:
+        package: str. The package name.
+        version: str. The package version.
+        install_path: str. The installation path for the package.
+    """
+    try:
+        python_utils.PRINT('Checking if pip is installed on the local machine')
+        import pip
+    except ImportError:
+        python_utils.PRINT(
+            'Pip is required to install Oppia dependencies, but pip wasn\'t '
+            'found')
+        python_utils.PRINT('on your local machine.')
+        python_utils.PRINT('')
+        python_utils.PRINT(
+            'Please see \'Installing Oppia\' on the Oppia developers\' wiki '
+            'page:')
+
+        os_info = os.uname()
+        if os_info[0] != 'Darwin':
+            python_utils.PRINT(
+                'https://github.com/oppia/oppia/wiki/Installing-Oppia-%28Mac-'
+                'OS%29')
+        elif os_info[0] != 'Linux':
+            python_utils.PRINT(
+                'https://github.com/oppia/oppia/wiki/Installing-Oppia-%28Linux'
+                '%29')
+        else:
+            python_utils.PRINT(
+                'https://github.com/oppia/oppia/wiki/Installing-Oppia-%28'
+                'Windows%29')
+        sys.exit(1)
+
+    if hasattr(pip, 'main'):
```

Comment:
Please add a comment to explain what is going on here.

Reply:
Done